diffusion/evaluation/clean_fid_eval.py

```diff
@@ -0,0 +1,237 @@
+                 precision: str = 'amp_fp16',
+                 prompts: Optional[List[str]] = None):
+        self.model = model
+        self.tokenizer: PreTrainedTokenizerBase = model.tokenizer
+        self.eval_dataloader = eval_dataloader
+        self.clip_metric = clip_metric
+        self.load_path = load_path
+        self.guidance_scales = guidance_scales if guidance_scales is not None else [1.0]
+        self.size = size
+        self.batch_size = batch_size
+        self.loggers = loggers
+        self.seed = seed
+        self.output_dir = output_dir
+        self.num_samples = num_samples if num_samples is not None else float('inf')
+        self.precision = precision
+        self.prompts = prompts if prompts is not None else ['A shiba inu wearing a blue sweater']
+
+        # Init loggers
+        if self.loggers and dist.get_local_rank() == 0:
+            for logger in self.loggers:
+                if isinstance(logger, WandBLogger):
+                    wandb.init(**logger._init_kwargs)
+
+        # Load the model
+        Trainer(model=self.model,
+                load_path=self.load_path,
+                load_weights_only=True,
+                eval_dataloader=self.eval_dataloader)
+
+        # Move CLIP metric to device
+        self.device = dist.get_local_rank()
+        self.clip_metric = self.clip_metric.to(self.device)
+
+        # Predownload the CLIP model for computing clip-fid
+        _, _ = clip.load('ViT-B/32', device=self.device)
+
+    def _generate_images(self, guidance_scale: float):
+        """Core image generation function. Generates images at a given guidance scale.
+
+        Args:
+            guidance_scale (float): The guidance scale to use for image generation.
+        """
+        # Verify output dirs exist, if they don't, create them
+        real_image_path = os.path.join(self.output_dir, f'real_images_gs_{guidance_scale}')
+        gen_image_path = os.path.join(self.output_dir, f'gen_images_gs_{guidance_scale}')
+        if not os.path.exists(real_image_path) and dist.get_local_rank() == 0:
+            os.makedirs(real_image_path)
+        if not os.path.exists(gen_image_path) and dist.get_local_rank() == 0:
+            os.makedirs(gen_image_path)
+
+        # Reset the CLIP metric
+        self.clip_metric.reset()
+
+        # Storage for prompts
+        prompts = {}
+        # Iterate over the eval dataloader
+        num_batches = len(self.eval_dataloader)
+        for batch_id, batch in tqdm(enumerate(self.eval_dataloader)):
+            # Break if enough samples have been generated
+            if batch_id * self.batch_size * dist.get_world_size() >= self.num_samples:
+                break
+
+            real_images = batch['image']
+            captions = batch['captions']
+            # Ensure a new seed for each batch, as randomness in model.generate is fixed.
+            starting_seed = self.seed + num_batches * dist.get_local_rank()
+            seed = starting_seed + batch_id
```

Comment:
Nit: What's the reason for `starting_seed`? Would this be the same behavior?
```suggestion
            starting_seed = self.seed + batch_id * dist.get_local_rank()
```

Reply:
`starting_seed` exists here to ensure that every image is generated using different randomness. This isn't the only way to ensure that though. If I understand correctly, the change you proposed here won't have the same effect, because different ranks will see the same seed, and hence generate the same random sequence. Example:

Current behavior:
```
seed = 17
num_batches = 3

batch_id   rank 0 seed   rank 1 seed
0            17               20                
1            18               21
2            19               22
```

Proposed behavior (if I understand the suggestion correctly):
```
seed = 17
num_batches = 3

batch_id   rank 0 seed   rank 1 seed
0            17               17                
1            17               21
2            17               22
```

Reply:
Ah, I needed `batch_id + 1` for it to work.